### DISCUSSION TITLE
Is `v -g` working yet?

### DISCUSSION THREAD

**Discussion by xiongmao86** [vlang/v#17779] [Questions and Answers] [answered]
I have a default project generate by `v new` and when I compile the project with 0.3.3 version `v -g .`, there is no debug information emitted. In linux I try to `llvm-dwarfdump a.out` shows no symbol information. Is `-g` actually works?

**Comment by JalonSolov** [comment, marked as answer]
First, V doesn't create `a.out` files (unless you tell it to with `-o a.out`), so I don't know why you would be looking at that.

Debug info is there by default.  In fact, you have to run the `strip` command to remove it.
```
$ v hello_world.v
$ file hello_world
hello_world: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 4.4.0, not stripped
$
```
(the `not stripped` at the end means there is debug info present)

Running `objdump -W hello_world` displays all the debug info (quite a bit of output).

As far as `-g` option, all it does is make it so you see V line numbers instead of C line numbers in error messages.  You can also use `-cg` for more complete info when debugging.

I don't know what the command you used is supposed to do, but it wasn't doing it correctly, or you were just looking at the wrong file.

**Reply by xiongmao86** [reply]
Hi, @JalonSolov, thank you very much for your help. 

> First, V doesn't create `a.out` files (unless you tell it to with `-o a.out`), so I don't know why you would be looking at that.

I mess up with the output of `clang++`, when I compare it with the output of `v`. But what I checked was the right file, the executive output of `v -g .`, **let's call it 'hello_world` for the rest of the post.**

> Running `objdump -W hello_world` displays all the debug info (quite a bit of output).

I ran `llvm-dwarfdump -a hello_world` again with added `-a` option, then I see info on `.eh_frame` content, I think the command is working correctly, `v -g .` actually generating debug information, but it's not the information I am talking about.

**Too long, don't read**
> As far as -g option, all it does is make it so you see V line numbers instead of C line numbers in error messages. You can also use -cg for more complete info when debugging.

I want to list source, step in or step over on ~~instructions~~ statements, and inspect `v` variable values. I think probably `v -g` doesn't provide such kind of debug information as you said all it does is show v line numbers instead of C line numbers. 

My intention was to use `VSCode` to debug `v` generated executive files, `v -g .` doesn't provide the kind of debug information I need, do I understand correctly?
